### DISCUSSION TITLE
Compatibility with openssl 1.1.1 and openssl 3.1.1

### DISCUSSION THREAD

**Discussion by NishaBhalke** [openssl/openssl#22219] [Q&A]
We built applications using the OPC UA protocol, it works on Client -Server communication model. 
So, we are using OpenSSL3.1.1 for Client and OpenSSL1.1.1 for Server..
I was not able to establish a successful connection. It's failing at EVP_PKEY_decrypt() method. 
Is openSSL1.1.1 comapatible with OPenssl3.1.1

**Comment by t8m** [comment]
1.1.1 should be fully compatible with 3.1.1

Most likely your application is depending on some undefined behavior or something like that. Is your application able to communicate if both ends use the same version i.e., 1.1.1 or 3.1.1?

**Reply by NishaBhalke** [reply]
Yes, with both client and server using the same version, they are able to communicate. Only when client and server are using different versions, there is an issue.

**Reply by t8m** [reply]
Could it be the type of padding used in EVP_PKEY_enc/decrypt?
